packages/zendesk-adapter/src/filters/guide_arrange_paths.ts

```diff
@@ -0,0 +1,210 @@
+  [ARTICLE_TRANSLATION_TYPE_NAME]: 'translations',
+  [ARTICLE_TYPE_NAME]: 'articles',
+  [CATEGORY_TYPE_NAME]: 'categories',
+  [SECTION_TYPE_NAME]: 'sections',
+  [SECTION_TRANSLATION_TYPE_NAME]: 'translations',
+  [CATEGORY_TRANSLATION_TYPE_NAME]: 'translations',
+  [GUIDE_SETTINGS_TYPE_NAME]: 'settings',
+  [USER_SEGMENT_TYPE_NAME]: 'user_segments',
+  [PERMISSION_GROUP_TYPE_NAME]: 'permission_groups',
+  [GUIDE_LANGUAGE_SETTINGS_TYPE_NAME]: 'language_settings',
+}
+
+/**
+ * calculates a path which is not related to a specific brand
+ */
+const pathForGlobalTypes = (instance: InstanceElement): readonly string[] | undefined => [
+  ...GUIDE_PATH,
+  GUIDE_ELEMENT_DIRECTORY[instance.elemID.typeName],
+  pathNaclCase(instance.elemID.name),
+]
+
+/**
+ * calculates a path which is related to a specific brand and does not have a parent
+ */
+const pathForBrandSpecificRootElements = (instance: InstanceElement, brandName: string)
+  : readonly string[] | undefined => {
+  if (brandName === undefined) {
+    return undefined
+  }
+  const newPath = [
+    ...GUIDE_PATH,
+    'brands',
+    brandName,
+    GUIDE_ELEMENT_DIRECTORY[instance.elemID.typeName],
+    pathNaclCase(instance.elemID.name),
+  ]
+  if (instance.elemID.typeName === CATEGORY_TYPE_NAME) { // each category has a folder of its own
+    newPath.push(pathNaclCase(instance.elemID.name))
+  }
+  return newPath
+}
+
+/**
+ * calculates a path which is related to a specific brand and has a parent.
+ */
+const pathForOtherLevels = (params :{
+  instance: InstanceElement
+  needTypeDirectory: boolean
+  needOwnFolder: boolean
+  parent: InstanceElement
+}): readonly string[] | undefined => {
+  const parentPath = params.parent?.path
+  if (parentPath === undefined) {
+    return parentPath
+  }
+  const newPath = parentPath.slice(0, parentPath.length - 1)
+  if (params.needTypeDirectory) {
+    newPath.push(GUIDE_ELEMENT_DIRECTORY[params.instance.elemID.typeName])
+  }
+  if (params.needOwnFolder) {
+    newPath.push(pathNaclCase(params.instance.elemID.name))
+  }
+  newPath.push(pathNaclCase(params.instance.elemID.name))
+  return newPath
+}
+
+const getId = (instance: InstanceElement): number =>
+  instance.value.id
+
+const getFullName = (instance: InstanceElement): string =>
+  instance.elemID.getFullName()
+
+/**
+ * This filter arranges the paths for guide elements.
+ */
+const filterCreator: FilterCreator = () => ({
+  onFetch: async (elements: Element[]): Promise<void> => {
+    const guideInstances = elements
+      .filter(isInstanceElement)
+      .filter(inst => Object.keys(GUIDE_ELEMENT_DIRECTORY).includes(inst.elemID.typeName))
+    const guideGrouped = _.groupBy(guideInstances, inst => inst.elemID.typeName)
+    const parents = guideInstances.filter(instance => PARENTS.includes(instance.elemID.typeName))
+    const parentsById = _.keyBy(parents, getId)
+    const NameByIdParents = _.mapValues(_.keyBy(parents, getFullName), 'value.id')
```

Comment:
1.
```suggestion
    const nameByIdParents = _.mapValues(_.keyBy(parents, getFullName), 'value.id')
```
2. same re making sure all elements have an id (we can also do one check in the beginning and omit all instances that don't have an id)